### DISCUSSION TITLE
Dependency Submission API doesn't allow arbitrary package URL `type`s

### DISCUSSION THREAD

**Discussion by jamietanna** [community/community#87501] [Code Security] [answered]
### Select Topic Area

Bug

### Body

As noted in https://github.com/package-url/purl-spec/issues/286, the  [GitHub's Dependency Submission API](https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28#create-a-snapshot-of-dependencies-for-a-repository), which consumes pURL(s) appears to be rejecting pURLs with arbitrary `type`s, which seems unexpected.

I've asked for clarification in https://github.com/package-url/purl-spec/issues/286 for the spec's purposes, and also raising here for visibility.

I would expect that the API would allow augmenting with _any_ package data, rather than requiring it just be "known" types.

As noted in https://github.com/package-url/purl-spec/issues/286#issue-2081869640, the 

(Aside: I believe that in the above case, a translation from `pkg:mix` -> `pkg:hex` would make sense)

However, there will still be package types that aren't "known" - for instance organisation-specific ones - and therefore restricting usage of them in the API is a little frustrating.

**Comment by Naimulnobel** [comment]
It seems like you're encountering an issue with GitHub's Dependency Submission API rejecting package URLs (pURLs) with arbitrary types. This behavior might be unexpected if the pURL specification allows arbitrary types, but the API might have its own restrictions for various reasons, such as data consistency or security.

If you believe this is a bug or a limitation that should be addressed, the best course of action would be to report it to GitHub Support. They can provide more information about whether this behavior is intentional and if there are any plans to change it.

In the meantime, as a workaround, you might consider translating unknown package types to known ones, as you suggested with translating `pkg:mix` to `pkg:hex`. However, this might not be feasible for all package types, especially organization-specific ones.

Please note that this is a general advice based on the information you provided. For more specific advice, you might need to provide more details about your use case and the exact problems you're encountering.

**Comment by h4md153v63n** [comment]
If you are referring to a specific API or service related to package dependencies, it's important to consult the official documentation or support resources provided by the service or platform you are using. The capabilities and restrictions of an API, including whether arbitrary package URLs are allowed, would depend on the design and policies of that particular API.

**Reply by queenofcorgis** [reply]
@h4md153v63n 
We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by jamietanna** [comment, marked as answer]
This is now supported 👏🏽 https://github.blog/changelog/2025-04-03-dependency-graph-supports-all-purl-identified-package-ecosystems
